### PR TITLE
Historical dues and transaction data import

### PR DESCRIPTION
**Pull request by kberzinch**
Summary of changes:
- Adds `SquareCashTransaction` and `SquareTransaction` models and import commands for each
- Adds import command to map 2010-2017 dues spreadsheets to Apiary models
- Adds commands to match `SquareCashTransaction`s and `SquareTransaction`s to `Payment`s for backfilling all of the new fields that were added to `Payment`s in #1512 
- Adds additional error detection and better exception messages to `CreateOrUpdateUserFromBuzzAPI` job
- Add option to `CreateDuesPackages` Nova action to skip creating non-student package (intended for use with historical import only)

### CODE REVIEW

File: app/Models/Payment.php [hunk 1]
```diff
@@ -134,4 +139,29 @@ public function getRelationshipPermissionMap(): array
...
+        $this->created_at = $transaction->transaction_timestamp;
+        $this->save();
+    }
+
+    public function updateFromSquareTransaction(SquareTransaction $transaction): void
+    {
+        $this->method = 'swipe';  // note that some POS transactions are actually keyed for some reason
+        $this->amount = $transaction->amount;
+        $this->processing_fee = $transaction->processing_fee;
+        $this->created_at = $transaction->transaction_timestamp;
```
Comment (kberzinch) [inline app/Models/Payment.php:158]: It occured to me that we might want to add a `transaction_timestamp` to the `Payment` model in lieu of backdating these fields. That would also help with #830 somewhat.

File: app/Nova/Actions/CreateDuesPackages.php [hunk 2]
```diff
@@ -142,6 +144,10 @@ public function handle(ActionFields $fields, Collection $models): array
      */
     public function fields(): array
     {
-        return [];
+        return [
+            Boolean::make('Non-Student Package', 'non_student')
+                ->help('Whether to create a package for non-students')
```
Comment (JoshOldenburg) [inline app/Nova/Actions/CreateDuesPackages.php:149]:
```suggestion
                ->help('Whether to create a package for non-students. This should generally be left checked.')
```

**Review by JoshOldenburg — CHANGES_REQUESTED** [review]

File: app/Imports/HistoricalDues.php [hunk 3]
```diff
@@ -0,0 +1,452 @@
...
+            if (null === $user) {
+                CreateOrUpdateUserFromBuzzAPI::dispatchNow('gtid', $gtid, self::USER_CREATION_REASON_STRING);
+                $user = User::where('gtid', $gtid)->firstOrFail();
+            }
+        }
+
+        foreach ($packages as $package) {
+            $guessShirtSize = self::guessShirtSize($row);
+            if (null !== $guessShirtSize) {
+                $user->shirt_size = $guessShirtSize;
```
Comment (kberzinch) [inline app/Imports/HistoricalDues.php:114]: Is it more correct to leave `polo_size` blank or copy the value from `shirt_size`?

File: app/Jobs/CreateOrUpdateUserFromBuzzAPI.php [hunk 4]
```diff
@@ -136,6 +147,13 @@ public function handle(): void
         if ($user->is_service_account) {
             throw new Exception('BuzzAPI job attempted to create/update an account for an existing service account');
         }
+
+        foreach (self::EXPECTED_ATTRIBUTES as $attr) {
+            if (! property_exists($account, $attr)) {
+                throw new Exception('Selected account for '.$searchValue.' is missing expected attribute '.$attr);
+            }
+        }
```
Comment (kberzinch) [inline app/Jobs/CreateOrUpdateUserFromBuzzAPI.php:155]: This doesn't seem to work and I'm not sure why. The same code works in the import job.

File: app/Imports/HistoricalDues.php (hunk 3 above)
Comment (JoshOldenburg) [reply app/Imports/HistoricalDues.php:114]: Leave it blank

File: app/Models/Payment.php (hunk 1 above)
Comment (JoshOldenburg) [reply app/Models/Payment.php:159]: Agreed that this would be good eventually but it doesn't need to happen for this PR.

**Review by JoshOldenburg — APPROVED** [review]

**Pull request merged**